update tidb_hash_join_version (#20308)
pingcap/docs: system-variables.md

### tidb_hash_join_version <span class="version-mark">New in v8.4.0</span>

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): Yes
- Type: Enumeration

> - Default value: `optimized`

Reviewer: ```suggestion
- Default value: `optimized`. Before v9.0.0, the default value is `legacy`.
```

- Possible values: `legacy`, `optimized`

> - This variable is used to control whether TiDB uses an optimized version of hash join. If it is set to `optimized`, TiDB uses the optimized version to execute hash join for better performance.

Reviewer 2: ```suggestion
- This variable is used to control whether TiDB uses the [optimized version of hash join](/sql-statements/sql-statement-explain-analyze.md#hashjoinv2). If it is set to `optimized`, TiDB uses the optimized version to execute hash join for better performance.
```

Reviewer 2: ref: https://github.com/pingcap/docs/pull/20523/files

> **Note:**
>

> > Currently, the optimized hash join only supports inner join, outer join, semi join and anti semi join, so for other joins, even if `tidb_hash_join_version` is set to `optimized`, TiDB still uses the legacy hash join.

Reviewer: ```suggestion
> Currently, the optimized hash join only supports inner join, outer join, semi join and anti semi join. For other joins, even if `tidb_hash_join_version` is set to `optimized`, TiDB still uses the `legacy` hash join.
```

> > Currently, the optimized hash join only supports inner join, outer join, semi join and anti semi join. For other joins, even if `tidb_hash_join_version` is set to `optimized`, TiDB still uses the `legacy` hash join.

Reviewer 2: ```suggestion
> Currently, the optimized hash join only supports inner join, outer join, semi join, and anti semi join. For other joins, even if `tidb_hash_join_version` is set to `optimized`, TiDB still uses the `legacy` hash join.
```
